x/feedistribution/keeper/query.go

```diff
@@ -1,7 +1,227 @@
+	}
+	return &types.QueryAVSRewardAssetResponse{AvsRewardAsset: assetInfo}, nil
+}
+
+// RewardAssetsByAVS queries all reward assets for an AVS.
+func (k Keeper) RewardAssetsByAVS(ctx context.Context, req *types.AVSRequest) (*types.QueryRewardAssetsByAVSResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+	c := sdk.UnwrapSDKContext(ctx)
+	allRewardAssets, err := k.GetAllRewardAssetsByAVS(c, strings.ToLower(req.Avs))
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryRewardAssetsByAVSResponse{RewardAssets: &allRewardAssets}, nil
+}
+
+// AVSRewardParam queries the reward param for an AVS.
+func (k Keeper) AVSRewardParam(ctx context.Context, req *types.AVSRequest) (*types.QueryAVSRewardParamResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+	c := sdk.UnwrapSDKContext(ctx)
+	rewardParam, err := k.GetAVSRewardParam(c, req.Avs)
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryAVSRewardParamResponse{AvsRewardParam: rewardParam}, nil
+}
+
+// AVSCommunityPool queries the community reward pool for an AVS.
+func (k Keeper) AVSCommunityPool(ctx context.Context, req *types.AVSRequest) (*types.QueryAVSCommunityPoolResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+	c := sdk.UnwrapSDKContext(ctx)
+	feePool, err := k.GetAVSFeePool(c, req.Avs)
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryAVSCommunityPoolResponse{FeePool: &feePool}, nil
+}
+
+// AVSRewardDistribution queries the distribution information for an AVS.
+func (k Keeper) AVSRewardDistribution(ctx context.Context, req *types.AVSRequest) (*types.QueryAVSRewardDistributionResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+	c := sdk.UnwrapSDKContext(ctx)
+	avsRewardDistribution, err := k.GetAVSRewardDistribution(c, req.Avs)
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryAVSRewardDistributionResponse{AvsRewardDistribution: avsRewardDistribution}, nil
+}
+
+// OperatorOutstandingRewards queries the outstanding rewards for an operator.
+func (k Keeper) OperatorOutstandingRewards(ctx context.Context, req *types.OperatorAVSRequest) (*types.QueryOperatorOutstandingRewardsResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+	c := sdk.UnwrapSDKContext(ctx)
+	outstandingRewards, err := k.GetOperatorOutstandingRewards(c, req.Operator, strings.ToLower(req.Avs))
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryOperatorOutstandingRewardsResponse{OperatorOutstandingRewards: &outstandingRewards}, nil
+}
+
+// StakerOutstandingRewards queries the outstanding rewards for a staker.
+func (k Keeper) StakerOutstandingRewards(ctx context.Context, req *types.QueryStakerOutstandingRewardsRequest) (*types.QueryStakerOutstandingRewardsResponse, error) {
+	if req == nil {
```

Comment:
only verify the req, but not for the staker address ? maybe we can add verification for staker like 

```
func (k Keeper) validateStakerAddress(addr string) error {
    if addr == "" {
        return status.Error(codes.InvalidArgument, "staker address cannot be empty")
    }
    if _, err := sdk.AccAddressFromBech32(addr); err != nil {
        return status.Error(codes.InvalidArgument, fmt.Sprintf("invalid staker address: %v", err))
    }
    return nil
}
```